### PR TITLE
Some build refactors/improvements

### PR DESCRIPTION
**Pull request by timmywil**
### Summary ###

- migrate more uses of fs.promises; use node: protocol. I had missed a lot because my vs code search index was off. I used git grep instead.
- fix size comparison for slim files. I fixed this by preferring `+slim.` to `+` in the regex. It was only matching on the latter even for the slim file, which made it so comparisons between a dirty build and the saved branch were always off by 5-6 bytes.
- make compare size cache readable for manual edits. This is in lieu of more complicated options added to the build command. It's not hard to just edit the JSON, but it might as well be readable.
- Fix worker restarts for when a browser fails the acknowledgement test (I just broke this in #5438)

### Checklist ###

* [x] New tests have been added to show the fix or feature works
* [x] If needed, a docs issue/PR was created at https://github.com/jquery/api.jquery.com

### CODE REVIEW

File: build/release/cdn.js [hunk 1]
```diff
@@ -49,7 +49,7 @@ function makeReleaseCopies( Release ) {
 	].forEach( ( { filesMap, cdnFolder } ) => {
 		shell.mkdir( "-p", cdnFolder );
 
-		Object.keys( filesMap ).forEach( key => {
+		Object.keys( filesMap ).forEach( ( key ) => {
```
Comment (timmywil) [inline build/release/cdn.js:52]: Ran prettier on this file

File: build/release/cdn.js [hunk 2]
```diff
@@ -60,28 +60,33 @@ function makeReleaseCopies( Release ) {
...
 async function makeArchives( Release ) {
-
 	Release.chdir( Release.dir.repo );
 
 	async function makeArchive( { cdn, filesMap, cdnFolder } ) {
 		return new Promise( ( resolve, reject ) => {
 			if ( Release.preRelease ) {
-				console.log( "Skipping archive creation for " + cdn + "; this is a beta release." );
+				console.log(
+					"Skipping archive creation for " + cdn + "; this is a beta release."
```
Comment (mgol) [inline build/release/cdn.js:88]: Let's use template literals for such things:
```suggestion
					`Skipping archive creation for ${ cdn }; this is a beta release.`
```

File: build/tasks/compare_size.mjs [hunk 3]
```diff
@@ -60,7 +60,9 @@ function cacheResults( results ) {
 }
 
 function saveCache( loc, cache ) {
-	return fs.promises.writeFile( loc, JSON.stringify( cache ) );
+
+	// Keep cache readable for manual edits
+	return fs.writeFile( loc, JSON.stringify( cache, null, "  " ) );
```
Comment (mgol) [inline build/tasks/compare_size.mjs:65]: Let's also add a trailing new line

File: build/release/cdn.js [hunk 4]
```diff
@@ -60,28 +60,33 @@ function makeReleaseCopies( Release ) {
...
-				text = fs.readFileSync( builtFile, "utf8" )
-					.replace( /"file":"([^"]+)"/,
-						"\"file\":\"" + unpathedFile.replace( /\.min\.map/, ".min.js\"" ) )
-					.replace( /"sources":\["([^"]+)"\]/,
-						"\"sources\":[\"" + unpathedFile.replace( /\.min\.map/, ".js" ) + "\"]" );
+				text = fs
+					.readFileSync( builtFile, "utf8" )
+					.replace(
+						/"file":"([^"]+)"/,
+						"\"file\":\"" + unpathedFile.replace( /\.min\.map/, ".min.js\"" )
```
[the remark this answers is not in the document]
Comment (timmywil) [reply build/release/cdn.js:67]: I'm fine making this a template literal, but I worry about changing the logic here and would require testing releases. How about we change that with the release PR?

[the remark this answers is not in the document]
Comment (mgol) [reply build/release/cdn.js]: Sure, that works!

**Review by mgol — APPROVED** [review]
LGTM. The comment:
https://github.com/jquery/jquery/pull/5440#discussion_r1519002065
is yet to be addressed but that's going to happen in a separate release PR.

**Comment by EricMujjona** [comment]
#

**Comment by EricMujjona** [comment]
```
``
- - [ ] # > ````~~@mmonteleone ~~`

**Pull request merged**